src/OowpQuery.php

```diff
@@ -107,6 +107,11 @@ public function sortByIds($ids) {
 	public function &get_posts() {
 		parent::get_posts();
 
+        if(isset($this->query['fields']) && ($this->query['fields'] != "all")){
+            //just the ids have been requested (or 'id=>parent' objects), rather than post objects
+            return $this->posts;
```

Comment:
This will result in a different return type, so please update the function doc block

A reference to documentation would be handy too: https://developer.wordpress.org/reference/classes/wp_query/#return-fields-parameter